kernel/arch/dreamcast/hardware/biosfont.c

```diff
@@ -70,20 +75,26 @@ int bfont_set_32bit_mode(int on) {
     return rv;
 }
 
-/* A little assembly that grabs the font address */
-extern uint8* get_font_address(void);
-__asm__("	.text\n"
-        "	.align 2\n"
-	".globl _get_font_address\n"
-        "_get_font_address:\n"
-        "	mov.l	syscall_b4,r0\n"
-        "	mov.l	@r0,r0\n"
-        "	jmp	@r0\n"
-        "	mov	#0,r1\n"
-        "\n"
-        "	.align 4\n"
-        "syscall_b4:\n"
-        "	.long	0x8c0000b4\n");
+/* From cdrom.c */
+extern mutex_t _g1_ata_mutex;
+
+int lock_bfont(void) {
+    if(mutex_lock(&_g1_ata_mutex) == -1) return -1;
```

Comment:
I just wanted to add in that I'm still not certain that this is correct. I authored the piece that this was originally in, but it was as an independent mutex intended to be the KOS side guarantor of the BIOS lock syscall. It shouldn't cause problems where there weren't already (the only time things would fail with an error are times when there would already be a failure) but when we finally work out non-blocking CD access this may need revisiting/adjusting to shrink down its scope.

No changes requested, just comments for posterity (as I'm likely to forget :P )